### PR TITLE
ci: fix crosscompile issue

### PR DESCRIPTION
**Pull request by troian**
ledger uses github.com/zondax/hid which includes some C files
in order to get ledger detected akash has to deal with CGO
and crosscompilation

fix is using Goreleaser packed in
https://github.com/troian/golang-cross Docker image
with bunch of toolchains for cross-compilation

Define goreleaser build flags in makefile. All goreleaser runs should be now done only through make

Use goreleaser to build test image

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -19,21 +19,25 @@ jobs:
...
-      - if: startsWith(github.ref,'refs/tags/v') && github.repository == 'ovrclk/akash'
-        name: docker login
-        run: echo '${{secrets.DOCKERHUB_PASSWORD}}' | docker login --username '${{secrets.DOCKERHUB_USERNAME}}' --password-stdin
+        run: |-
+          docker run \
+          --priviledged \
+          --env MAINNET=${MAINNET} \
+          -v /var/run/docker.sock:/var/run/docker.sock \
+          -v `pwd`:/go/src/github.com/ovrclk/akash \
+          goreng/golang-cross:latest \
```
Comment (Ropes) [inline .github/workflows/release.yaml:29]: Can you pick out the version which `latest` points to? `latest` is always asking for problems later.

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -19,21 +19,25 @@ jobs:
...
+        run: |-
+          docker run \
+          --priviledged \
+          -e MAINNET=${MAINNET} \
+          -e DOCKER_USERNAME=${{secrets.DOCKERHUB_USERNAME}} \
+          -e DOCKER_PASSWORD=${{secrets.DOCKERHUB_PASSWORD}} \
+          -e GITHUB_TOKEN=${{ secrets.GORELEASER_ACCESS_TOKEN }} \
+          -v /var/run/docker.sock:/var/run/docker.sock \
+          -v `pwd`:/go/src/github.com/ovrclk/akash \
+          goreng/golang-cross:latest \
```
Comment (Ropes) [inline .github/workflows/release.yaml:42]: `latest` -> semver tag

**Review by Ropes — APPROVED** [review]
Nice debugging the solution! 

I'd  change the tag from `latest` to save some headache down the road, otherwise looks good.

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -11,29 +11,27 @@ on:
...
+      - run: make modvendor
       - name: release dry-run
-        uses: goreleaser/goreleaser-action@v1
-        with:
-          args: --skip-validate --skip-publish
-          version: v0.133.0
-
-      - if: startsWith(github.ref,'refs/tags/v') && github.repository == 'ovrclk/akash'
-        name: docker login
-        run: echo '${{secrets.DOCKERHUB_PASSWORD}}' | docker login --username '${{secrets.DOCKERHUB_USERNAME}}' --password-stdin
```
Comment (boz) [inline .github/workflows/release.yaml:31]: are we still publishing images?

**Review by boz — COMMENTED** [review]
awesome, thank you.   question inline.

File: .github/workflows/release.yaml (hunk 3 above)
Comment (troian) [reply .github/workflows/release.yaml:31]: yah, `goreleaser` in `goreng/golang-cross` is publishing it

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -9,31 +9,28 @@ on:
       - v*
 
 env:
-  # bleh
-  MAINNET: ${{ github.ref == 'refs/heads/mainnet/main' || (startsWith(github.ref,'refs/tags/v') && ( contains(github.ref,'.0.') || contains(github.ref,'.2.') || contains(github.ref,'.4.') || contains(github.ref,'.6.') || contains(github.ref,'.8.') )) }}
```
Comment (boz) [inline .github/workflows/release.yaml:13]: thank you thank you thank you

File: .github/workflows/release.yaml [hunk 5]
```diff
@@ -9,31 +9,28 @@ on:
...
 jobs:
   goreleaser:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - run: git fetch --prune --unshallow
+      - name: branch name
+        id: branch_name
+        run: |
+          echo ::set-output name=MAINNET::$([[ $(./script/semver.sh get minor ${GITHUB_REF#refs/tags/) % 2 ]] && echo true || echo false)
```
Comment (boz) [inline .github/workflows/release.yaml:22]: what is this wizardry

File: .goreleaser.yaml [hunk 6]
```diff
@@ -1,27 +1,74 @@
...
-      - goos: darwin
-        goarch: 386
+    goarch:
+      - amd64
+    goos:
+      - windows
+    env:
+      - CGO_ENABLED=1
+      - CC=x86_64-w64-mingw32-gcc
+      - CXX=x86_64-w64-mingw32-g++
```
Comment (boz) [inline .goreleaser.yaml:58]: do you know if ledger works on windows?

**Review by boz — APPROVED** [review]
:rocket: :pray:

File: .github/workflows/release.yaml (hunk 5 above)
Comment (troian) [reply .github/workflows/release.yaml:22]: 💫

File: .goreleaser.yaml (hunk 6 above)
Comment (troian) [reply .goreleaser.yaml:58]: indeed
```
Y:\go\src\github.com\ovrclk\akash\dist\akashctl-windows-amd64_windows_amd64>akashctl.exe keys add testwin --ledger --index 5

- name: testwin
  type: ledger
  address: akash1y8za8kvk3cr7wc9wp9prglvd3ugkhd7vjk4254
  pubkey: akashpub1addwnpepqtzryyl3zsm0znv5rrc9zcu3c6549slc3ckmp30sr59ahwu5y947kydhhzz
  mnemonic: ""
  threshold: 0
  pubkeys: []
Y:\go\src\github.com\ovrclk\akash\dist\akashctl-windows-amd64_windows_amd64>
```

**Pull request merged**